Reworked element naming

elsoazemelet opened the pull request:
Closes #1232

See original ticket for feature definition.

Element name detection and changing is reworked.

- When an element name is changed via the new Element Name field:
    - If no Element Name block is present on setup event it is added. The value of the input field and the block should synchronously update each other
    - If Element Name block is present, it is updated
- When an element name is changed via the Element Name block:
    - If newly added manually, the name should update on the input field
    - If already present, changing it should update the input field
    - If deleted, or moved from first position, the input field resets

**NOTE:** Previous grid to editor element name runtime calculation is ignored (and is not used anymore)

elsoazemelet committed:
Reworked element naming

elsoazemelet committed:
Minor fixes

SukuWc committed:
SUKU layout improved

Greg-Orca commented:
The element name is not cleared/changed after:
- delete the setup>element name action block
- clear the element name
- only add a "space" character

elsoazemelet committed:
Fixed QA issue

Greg-Orca commented:
<img width="413" height="246" alt="Screenshot 2025-08-27 at 11 37 34" src="https://github.com/user-attachments/assets/46813f9e-d564-475e-bdaf-f73c22c5a889" />
The element name is not cleared if I delete the element name block,

elsoazemelet committed:
Fixed QA issue

Greg-Orca commented:
<img width="308" height="98" alt="Screenshot 2025-08-28 at 11 25 15" src="https://github.com/user-attachments/assets/6a6d14ef-f1f5-48a9-b76f-45eba35c43a9" />
please add a test id for this input field

Greg-Orca committed:
tests added

SukuWc committed:
Merge branch 'stable' into feat/ElementNaming

elsoazemelet committed:
Fixed test for element name

elsoazemelet committed:
Refixed tests

elsoazemelet commented:
> <img alt="Screenshot 2025-08-28 at 11 25 15" width="308" height="98" src="https://private-user-images.githubusercontent.com/88966111/483076313-6a6d14ef-f1f5-48a9-b76f-45eba35c43a9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY3NDc2MjAsIm5iZiI6MTc1Njc0NzMyMCwicGF0aCI6Ii84ODk2NjExMS80ODMwNzYzMTMtNmE2ZDE0ZWYtZjFmNS00OGE5LWI3NmYtNDVlYmEzNWM0M2E5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTAxVDE3MjIwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjOTk4N2I1Y2I5ZjdlY2U0MTMyMjE2NTU2OGUwY2Q4MGI4MjRmNDQzMjQ5YTRiZTRlZDAwZWMzOGMzZmUzNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.51msAGJRyHh0cO-urmAXvfX79x8git9wGi6v1tFOKtk"> please add a test id for this input field

Added ID, fixed locator. Some tests still are broken, can you please check on them? @Greg-Orca

Greg-Orca commented:
please use data-testid instead of data-test-id

https://playwright.dev/docs/locators#locate-by-test-id

elsoazemelet committed:
Fixed tests
